mypy.ini

```diff
@@ -0,0 +1,1070 @@
+
+[mypy-astropy.coordinates.builtin_frames.baseradec]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.builtin_frames.cirs_observed_transforms]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.builtin_frames.ecliptic]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.builtin_frames.ecliptic_transforms]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.builtin_frames.equatorial]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.builtin_frames.galactic]
+disable_error_code = assignment,attr-defined
+
+[mypy-astropy.coordinates.builtin_frames.galactocentric]
+disable_error_code = assignment,attr-defined,return-value,var-annotated
+
+[mypy-astropy.coordinates.builtin_frames.hadec]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.builtin_frames.icrs_observed_transforms]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.builtin_frames.intermediate_rotation_transforms]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.builtin_frames.itrs]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.builtin_frames.itrs_observed_transforms]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.builtin_frames.lsr]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.builtin_frames.skyoffset]
+disable_error_code = var-annotated
+
+[mypy-astropy.coordinates.builtin_frames.supergalactic]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.builtin_frames.utils]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.calculation]
+disable_error_code = import-untyped,var-annotated
+
+[mypy-astropy.coordinates.earth]
+disable_error_code = assignment
+
+[mypy-astropy.coordinates.earth_orientation]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.erfa_astrom]
+disable_error_code = import-untyped,var-annotated
+
+[mypy-astropy.coordinates.funcs]
+disable_error_code = attr-defined,import-untyped,var-annotated
+
+[mypy-astropy.coordinates.matching]
+disable_error_code = attr-defined,import-untyped
+
+[mypy-astropy.coordinates.polarization]
+disable_error_code = arg-type,assignment,misc
+
+[mypy-astropy.coordinates.representation.base]
+disable_error_code = attr-defined,var-annotated
+
+[mypy-astropy.coordinates.representation.cartesian]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.representation.geodetic]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.representation.spherical]
+disable_error_code = import-untyped
+
+[mypy-astropy.coordinates.sky_coordinate]
+disable_error_code = attr-defined,import-untyped
+
+[mypy-astropy.coordinates.solar_system]
+disable_error_code = attr-defined,import-not-found,import-untyped
+
+[mypy-astropy.coordinates.spectral_coordinate]
+disable_error_code = attr-defined
+
+[mypy-astropy.coordinates.spectral_quantity]
+disable_error_code = assignment,attr-defined
+
+[mypy-astropy.coordinates.tests.accuracy.generate_ref_ast]
```

Comment:
Do we want to type check the test files? I generally don't in my own code.

Reply:
me neither, but I think it's a good idea nevertheless: test code is a good source of user code, checking it might reveal that our type annotations are too strict compared to what we actually support and test

Reply:
Alternatively for the tests we could turn on beartype and do run-time checking of Astropy, checking all the type annotations without having to add type hints to the test suite.

```
from beartype.claw import beartype_this_package       # <-- hype comes
beartype_this_package() 
```

Reply:
I don't know that we need to add type annotations to tests but would definitely be good to have a way to have the tests actually test the annotations optionally.

Reply:
The bear-type route in the tests will accomplish this (testing the type hints without having to type the tests) and should only add a few lines to the test code + a test dependency. This also has the added benefit of testing runtime type checking! Which some users will definitely want.

Reply:
Interesting to hear about beartype!  

For PlasmaPy, my current thinking is to include type hint annotations for tests since they can sometimes really help with test readability and maintainability...especially for when we're using `@pytest.mark.parametrize` or are somewhat complicated.  I would be in favor of including type hint annotations for tests as well, but probably at a lower priority than for the application code.

Reply:
Adding type annotations to tests can be worthwhile if it helps IDEs offer better autocompletion.